### ISSUE TITLE
Check new Global with old Zone to handle top-level targetRef syntactic sugar properly

### ISSUE THREAD

**Issue by lobkovilya** [kumahq/kuma#11426]
### What happened?

1. Create policy on Global that doesn't have top-level targetRef
2. Let the policy to be synced to old zone (2.8.x)

**Comment by Automaat** [comment]
Zone will send NACK on this policy and we will see log: 
```
kds-delta-zone    received resource is invalid, sending NACK    {"kds-version": "v2", "err": "spec.targetRef.kind: must be defined"}
```

**Comment by Automaat** [comment]
> Does the issue fix itself when Zone CP is updated?
yes, it is synced properly after upgrading zone

> Does the KDS sync work for other valid policies of the same type?
yes, after applying another meshtimeout with explicit topLevel targetRef, it is synced correctly

One thing with default topLevel targetRef is that when running `kubectl get meshtimeouts -n kuma-system` we have missing targetRef kind in output:

![Image](https://github.com/user-attachments/assets/3a83d4a3-af35-46bc-a88c-fa2eae17c8f2)

**Comment by lobkovilya** [comment]
I tried:
1. Create 2 policies on Global with and without top-level targetRef
2. Connect Zone CP 2.8.3
3. None of the policies is synced to Zone CP (even the valid one)

Moreover, when I deleted the policy without top-level targetRef from Global, the valid policy is still not synced until I restart the Zone CP. 

I think we should put this syntactic sugar behind the KDS feature flag. If we have an option to avoid NACKs I'd rather go for it.
